analyzers/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldContainAssertion.cs

```diff
@@ -98,6 +95,36 @@ protected override void Initialize(SonarAnalysisContext context)
                 SyntaxKind.MethodDeclaration);
         }
 
+        private static bool ContainsAssertion(MethodDeclarationSyntax methodDeclaration, SemanticModel previousSemanticModel, ISet<IMethodSymbol> visitedSymbols, int level)
```

Comment:
hasAnyAssertion part should be also part of ContainsAssertion. Because this way we do not recognize things like x.Should().Be(y) if it is in a helper method.

Reply:
Also, now that the assert exception part is implemented, it would be nice to rebase on top of master and put hte hasAnyThrownAssertionExceptions  part inside the ContainsAssertion check too.

Reply:
Nice catch! I can simplify the logic and remove redundant `DescendantNodes()` evaluation (that can be expensive).